src/snowcli/cli/project/definition.py

```diff
@@ -0,0 +1,80 @@
+from .util import clean_identifier, get_env_username
+from pathlib import Path
+from typing import List
+from strictyaml import (
+    YAML,
+    load,
+    as_document,
+)
+
+from .schema import (
+    project_schema,
+    project_override_schema,
+)
+
+from snowcli.cli.common.snow_cli_global_context import snow_cli_global_context_manager
+
+DEFAULT_USERNAME = "unknown_user"
+
+
+def merge_left(target: dict | YAML, source: dict | YAML) -> None:
```

Comment:
```suggestion
def merge_left(target: Union[Dict, YAML], source: Union[Dict, YAML]) -> None:
```
Consider introducing a custom type for `Union[Dict, YAML]`